src/PIL/Jpeg2KImagePlugin.py

```diff
@@ -200,13 +200,21 @@ def _open(self):
                 "jpeg2k",
                 (0, 0) + self.size,
                 0,
-                (self.codec, self.reduce, self.layers, fd, length),
+                (self.codec, self._reduce, self.layers, fd, length),
             )
         ]
 
+    @property
+    def reduce(self):
+        return self._reduce or super().reduce
+
+    @reduce.setter
+    def reduce(self, value):
+        self._reduce = value
+
     def load(self):
-        if self.reduce:
-            power = 1 << self.reduce
+        if self._reduce:
+            power = 1 << self._reduce
             adjust = power >> 1
             self._size = (
                 int((self.size[0] + adjust) / power),
```

Comment:
This should be moved inside `if self.tile:` clause to prevent this bug: https://github.com/python-pillow/Pillow/pull/4344#issuecomment-599743249